src/org/rascalmpl/library/ParseTree.rsc

```diff
@@ -508,12 +508,75 @@ the cost of constructing nested ambiguity clusters.
 
 If the input sentence is not ambiguous after all, simply the entire tree is returned.
 }
-public Tree firstAmbiguity(type[Tree] begin, str input)
+Tree firstAmbiguity(type[Tree] begin, str input)
   = parser(begin, firstAmbiguity=true)(input, |unknown:///|);
 
-public Tree firstAmbiguity(type[Tree] begin, loc input)
+Tree firstAmbiguity(type[Tree] begin, loc input)
   = parser(begin, firstAmbiguity=true)(input, input);
 
+@javaClass{org.rascalmpl.library.Prelude}
+@synopsis{Generate a parser and store it in serialized form for later reuse.}
+@description{
+The stored parsers would be able to be recovered later using ((loadParsers)).
+
+For any concrete grammar, a.k.a. reified syntax type, `g` it holds that:
+* after `storeParsers(g, file);`
+* then `g = loadParsers(file);`
+* and given `h = parsers(g);`
+* then for all valid `nt`, `input` and `origin`: `g(nt, input, origin) == g(nt, input, origin)`
```

Comment:
both side of the `==` is the exact same?

Reply:
100% exact.

Reply:
ah. `h`